src/Model/Behavior/SlugBehavior.php

```diff
@@ -298,6 +303,55 @@ public function slug($entity, $string = null, $separator = null)
         return $slug;
     }
 
+    /**
+     * Gets the slug string based on an entity
+     *
+     * @param \Cake\Datasource\EntityInterface $entity Entity
+     * @param string $separator Separator
+     * @return string
+     */
+    protected function _getSlugStringFromEntity($entity, $separator)
+    {
+        $string = [];
+        foreach ((array)$this->getConfig('displayField') as $field) {
+            if ($entity->getError($field)) {
+                throw new InvalidArgumentException();
+            }
+            $string[] = $value = Hash::get($entity, $field);
+        }
+
+        return implode($separator, $string);
+    }
+
+    /**
+     * Builds the conditions
+     *
+     * @param \Cake\ORM\Entity $entity Entity.
+     * @param string $slug Slug
+     * @return array
+     */
+    protected function _conditions($entity, $slug)
+    {
+        /** @var string $primaryKey */
+        $primaryKey = $this->_table->getPrimaryKey();
+        $field = $this->_table->aliasField($this->getConfig('field'));
+
+        $conditions = [$field => $slug];
+
+        if (is_callable($this->getConfig('scope'))) {
+            $scope = $this->getConfig('scope');
+            $conditions += $scope($entity);
+        } else {
+            $conditions += $this->getConfig('scope');
+        }
+
+        if ($id = $entity->{$primaryKey}) {
```

Comment:
Move the assignment out of the `if` please.

Reply:
This has already been there before so I kept it.

Reply:
Yes I am aware of that, but since we are improving things let's take of this too.